File: 415.AddStrings/add.py
class Solution(object):
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        up = 0
        p1, p2 = len(num1)-1, len(num2)-1
        ans = ''
        while p1 >= 0 or p2 >= 0:
            v1, v2 = 0, 0
            if p1 >= 0:
                v1 = ord(num1[p1]) - ord('0')
                p1 -= 1
            if p2 >= 0:
                v2 = ord(num2[p2]) - ord('0')
                p2 -= 1
            s = v1 + v2 + up
            up = s / 10
            ans = chr(s % 10 + ord('0')) + ans

        ans = chr(up + ord('0')) + ans if up else ans

        return ans

if __name__ == '__main__':
    num1 = '999'
    num2 = '1'
    print Solution().addStrings(num1, num2)

File: 415.AddStrings/add.cc
class Solution {
public:
    string addStrings(string num1, string num2) {
        int p1 = num1.length() - 1, p2 = num2.length() - 1;
        int up = 0;
        string ans = "";
        while (p1 >= 0 || p2 >= 0) {
            int v1 = p1 >= 0 ? num1[p1--] - '0' : 0;
            int v2 = p2 >= 0 ? num2[p2--] - '0' : 0;
            int sum = v1 + v2 + up;
            up = sum / 10;
            ans = char(sum % 10 + '0') + ans;
        }
        return up > 0 ? char(up + '0') + ans : ans;
    }
};
